### PR TITLE
Add photos of seeds

### PR DESCRIPTION
**Pull request by Br3nda**

### CODE REVIEW

File: app/controllers/photos_controller.rb [hunk 1]
```diff
@@ -6,13 +6,14 @@ class PhotosController < ApplicationController
...
   def show
+    @crops = Crop.distinct.joins(:photographings).where(photographings: { photo: @photo})
     respond_with(@photo)
   end
 
   def index
-    if params[:crop_id]
-      @crop = Crop.find params[:crop_id]
-      @photos = @crop.photos
+    if params[:crop_slug]
```
Comment (Br3nda) [inline app/controllers/photos_controller.rb]: this is a bug fix

File: app/views/plantings/_progress.html.haml [hunk 2]
```diff
@@ -1,14 +1 @@
-- if planting.crop.perennial
-  = render "plantings/progress_bar", status: "perennial", progress: nil
```
Comment (Br3nda) [inline app/views/plantings/_progress.html.haml]: Moved all this to a helper

File: app/views/crops/show.html.haml [hunk 3]
```diff
@@ -19,69 +19,45 @@
 
 .row
   .col-md-9
-    .row
-      .col-md-12
+    %h2
```
Comment (Br3nda) [inline app/views/crops/show.html.haml]: Fixed the alignment of content - no longer hanging off to the left

File: app/views/crops/show.html.haml [hunk 4]
```diff
@@ -19,69 +19,45 @@
...
-        = will_paginate @posts, params: { anchor: "posts" }
-      - @posts.each do |post|
-        = render partial: "posts/single", locals: { post: post, subject: true }
-
-      .pagination
-        = page_entries_info @posts
-        = will_paginate @posts, params: { anchor: "posts" }
+    .row
+      .col-md-12
+        = render 'crops/posts', crop: @crop
```
Comment (Br3nda) [inline app/views/crops/show.html.haml]: Moved all the posts into their own partial

File: app/views/photos/show.html.haml [hunk 5]
```diff
@@ -27,3 +27,8 @@
     %p= link_to "View on Flickr", @photo.link_url
     - if @photo.associations?
       = render "photo_associations", locals: { photo: @photo }
+      
+    - if @crops.size.positive?
```
Comment (Br3nda) [inline app/views/photos/show.html.haml]: New feature: displaying the card for each of the crops in the photo

**Comment by Br3nda** [comment]
I've made the photo -> Crop relationship clearer. There's a new scope called Photo.by_crop, and also Photo.by_model_type to find photos.

This makes it tidier when looking for all photos for a crop, but also when viewing them by type of photo (a harvest versus seeds etc)

Photos of a garden do now have an associated crop.

**Comment by Br3nda** [comment]
Crop Page:
image

image


Photo page:
image
image

File: db/migrate/20190130090437_add_crop_to_photographings.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class AddCropToPhotographings < ActiveRecord::Migration[5.2]
+  def change
+    add_column(:photographings, :crop_id, :integer)
+    add_foreign_key(:photographings, :crops)
+
+    Photographing.where(crop_id: nil).each do |p|
```
Comment (Br3nda) [inline db/migrate/20190130090437_add_crop_to_photographings.rb]: I've checked and we have 2500 records in this table on line, which should be fine for this line of code to cope with

**Comment by Br3nda** [comment]
Anyone able to review?

**Comment by pozorvlak** [comment]
It's late here, but I'll try to take a look tomorrow.

**Pull request merged**